src/angular/schematics/ng-add/migrations/menu-migration.ts

```diff
@@ -0,0 +1,153 @@
+import { dirname } from '@angular-devkit/core';
+import {
+  addModuleImportToModule,
+  DevkitContext,
+  Migration,
+  ResolvedResource,
+  TargetVersion,
+} from '@angular/cdk/schematics';
+import { findModule } from '@schematics/angular/utility/find-module';
+
+import {
+  findReferenceAttribute,
+  iterateNodes,
+  MigrationElement,
+  MigrationRecorderRegistry,
+  nodeCheck,
+} from '../../utils';
+
+export class MenuMigration extends Migration<null, DevkitContext> {
+  enabled: boolean = this.targetVersion === ('merge' as TargetVersion);
+
+  private _contextmenus = new MigrationRecorderRegistry(this);
+  private _dropdowns = new MigrationRecorderRegistry(this);
+  private _referenceNamesPerTemplate = new Map<ResolvedResource, string[]>();
+
+  /** Method that will be called for each Angular template in the program. */
+  visitTemplate(template: ResolvedResource): void {
+    if (!this._referenceNamesPerTemplate.has(template)) {
+      this._referenceNamesPerTemplate.set(template, []);
+    }
+
+    iterateNodes(template.content, async (node) => {
+      if (nodeCheck(node).is('sbb-contextmenu')) {
+        this._contextmenus.add(template, node);
+      } else if (nodeCheck(node).is('sbb-dropdown')) {
+        this._dropdowns.add(template, node);
+      }
+
+      const referenceAttribute = findReferenceAttribute(node);
+      if (referenceAttribute) {
+        this._referenceNamesPerTemplate.get(template)!.push(referenceAttribute.name);
```

Comment:
It would be slightly more efficient to declare an array variable, instead of getting the array each time from _referenceNamesPerTemplate